Compilation warning with GROW R503 fingerprint reader component

### The problem

When I compile my code using GROW R503 I get multiple compilation warnings

### Which version of ESPHome has the issue?

2026.1.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

GROW R503

### YAML Config

```yaml
esphome:
  name: ${name}
  friendly_name: ${friendly_name}
  name_add_mac_suffix: false
  project:
    name: esphome.web
    version: '1.0'
  platformio_options:
    board_build.flash_mode: dio 

esp32:
  board: esp32-s3-devkitc-1
  variant: esp32s3
  flash_size: 4MB  
  framework:
    type: esp-idf
```

### Anything in the logs that might be useful for us?

```txt
Compiling .pioenvs/fingerprint/src/esphome/components/fingerprint_grow/fingerprint_grow.cpp.o
In file included from src/esphome/core/component.h:9,
                 from src/esphome/components/fingerprint_grow/fingerprint_grow.h:3,
                 from src/esphome/components/fingerprint_grow/fingerprint_grow.cpp:1:
src/esphome/components/fingerprint_grow/fingerprint_grow.cpp: In member function 'virtual void esphome::fingerprint_grow::FingerprintGrowComponent::dump_config()':
src/esphome/components/fingerprint_grow/fingerprint_grow.cpp:541:74: warning: 'virtual std::string esphome::GPIOPin::dump_summary() const' is deprecated: Override dump_summary(char*, size_t) instead. Will be removed in 2026.7.0. [-Wdeprecated-declarations]
  541 |                 this->has_sensing_pin_ ? this->sensing_pin_->dump_summary().c_str() : "None",
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/esphome/core/log.h:98:101: note: in definition of macro 'esph_log_config'
   98 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                     ^~~~~~~~~~~
src/esphome/components/fingerprint_grow/fingerprint_grow.cpp:535:3: note: in expansion of macro 'ESP_LOGCONFIG'
  535 |   ESP_LOGCONFIG(TAG,
      |   ^~~~~~~~~~~~~
In file included from src/esphome/core/hal.h:4,
                 from src/esphome/components/uart/uart.h:5,
                 from src/esphome/components/fingerprint_grow/fingerprint_grow.h:7:
src/esphome/core/gpio.h:145:20: note: declared here
  145 | inline std::string GPIOPin::dump_summary() const { return {}; }
      |                    ^~~~~~~
src/esphome/components/fingerprint_grow/fingerprint_grow.cpp:542:77: warning: 'virtual std::string esphome::GPIOPin::dump_summary() const' is deprecated: Override dump_summary(char*, size_t) instead. Will be removed in 2026.7.0. [-Wdeprecated-declarations]
  542 |                 this->has_power_pin_ ? this->sensor_power_pin_->dump_summary().c_str() : "None");
      |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/esphome/core/log.h:98:101: note: in definition of macro 'esph_log_config'
   98 |   ::esphome::esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                                     ^~~~~~~~~~~
src/esphome/components/fingerprint_grow/fingerprint_grow.cpp:535:3: note: in expansion of macro 'ESP_LOGCONFIG'
  535 |   ESP_LOGCONFIG(TAG,
      |   ^~~~~~~~~~~~~
src/esphome/core/gpio.h:145:20: note: declared here
  145 | inline std::string GPIOPin::dump_summary() const { return {}; }
      |                    ^~~~~~~
```

### Additional information

_No response_

--- a/esphome/components/fingerprint_grow/fingerprint_grow.cpp
+++ b/esphome/components/fingerprint_grow/fingerprint_grow.cpp
@@ -1,4 +1,5 @@
 #include "fingerprint_grow.h"
+#include "esphome/core/gpio.h"
 #include "esphome/core/log.h"
 #include <cinttypes>
 
@@ -532,14 +533,21 @@ void FingerprintGrowComponent::sensor_sleep_() {
 }
 
 void FingerprintGrowComponent::dump_config() {
+  char sensing_pin_buf[GPIO_SUMMARY_MAX_LEN];
+  char power_pin_buf[GPIO_SUMMARY_MAX_LEN];
+  if (this->has_sensing_pin_) {
+    this->sensing_pin_->dump_summary(sensing_pin_buf, sizeof(sensing_pin_buf));
+  }
+  if (this->has_power_pin_) {
+    this->sensor_power_pin_->dump_summary(power_pin_buf, sizeof(power_pin_buf));
+  }
   ESP_LOGCONFIG(TAG,
                 "GROW_FINGERPRINT_READER:\n"
                 "  System Identifier Code: 0x%.4X\n"
                 "  Touch Sensing Pin: %s\n"
                 "  Sensor Power Pin: %s",
-                this->system_identifier_code_,
-                this->has_sensing_pin_ ? this->sensing_pin_->dump_summary().c_str() : "None",
-                this->has_power_pin_ ? this->sensor_power_pin_->dump_summary().c_str() : "None");
+                this->system_identifier_code_, this->has_sensing_pin_ ? sensing_pin_buf : "None",
+                this->has_power_pin_ ? power_pin_buf : "None");
   if (this->idle_period_to_sleep_ms_ < UINT32_MAX) {
     ESP_LOGCONFIG(TAG, "  Idle Period to Sleep: %" PRIu32 " ms", this->idle_period_to_sleep_ms_);
   } else {

